optuna/visualization/_parallel_coordinate.py

```diff
@@ -167,6 +181,17 @@ def _target(t: FrozenTrial) -> float:
 
         dims.append(dim)
 
+    if len(numeric_cat_params.keys()) != 0:
+        sorted_categorical_vals = (
+            pd.DataFrame()
+            .from_dict(numeric_cat_params)
+            .sort_values(by=list(numeric_cat_params.keys()))
+        )
+        idx = sorted_categorical_vals.index
```

Comment:
Could you implement this sort of logic without `pandas` because pandas is an optional dependency? 

By the way, if this sort of logic depends on pandas, then we need to `try_import`. For example, 
https://github.com/optuna/optuna/blob/c6c083d46adfaf626b4ea81f7254f5bc60305fa0/optuna/study/_dataframe.py#L14-L16